<launch>
  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="raw_obstacle_laser_filter">
	  <rosparam command="load" file="$(find static_avoidance)/launch/box_filter/box_filter_raw.yaml" />
          <remap from="scan_filtered" to="raw_scan_filtered"/>
  </node>

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="left_obstacle_laser_filter">
	  <rosparam command="load" file="$(find static_avoidance)/launch/box_filter/box_filter_left.yaml" />
	  <remap from="scan_filtered" to="left_scan_filtered"/>
  </node>

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="right_obstacle_approach">
	  <rosparam command="load" file="$(find static_avoidance)/launch/box_filter/box_filter_right.yaml" />
	  <remap from="scan_filtered" to="right_scan_filtered"/>
  </node>
  

  <node pkg="static_avoidance" name="static_avoidance_car" type="static_avoidance" output="screen">
  </node>

  <include file="$(find lms1xx)/launch/LMS1xx.launch"/>
  
  <node name="obstacle_extractor_raw" pkg="obstacle_detector" type="obstacle_extractor_node">
    <remap from="scan" to="raw_scan_filtered"/>
    <remap from="raw_obstacles" to="raw_obstacles"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="false"/>
    <param name="discard_converted_segments" value="false"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="3"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="1.5"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.1"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>
   </node>
  <node name="obstacle_extractor_left" pkg="obstacle_detector" type="obstacle_extractor_node">
    <remap from="scan" to="left_scan_filtered"/>
    <remap from="raw_obstacles" to="raw_obstacles_left"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="false"/>
    <param name="discard_converted_segments" value="false"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="3"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="1.5"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.1"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>

  </node>
  
  <node name="obstacle_extractor_right" pkg="obstacle_detector" type="obstacle_extractor_node">
    <remap from="scan" to="right_scan_filtered"/>
    <remap from="scan_filtered" to="raw_scan_filtered"/>
    <remap from="raw_obstacles" to="raw_obstacles_right"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="false"/>
    <param name="discard_converted_segments" value="false"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="3"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="1.5"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.1"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>
  </node>

</launch>
